### PR TITLE
Allow indexing ints and index int primary keys

### PR DESCRIPTION
**Pull request by tgoyne**
The remaining core issues appear to have been solved.

@alazier @jpsim

### CODE REVIEW

File: Realm/Tests/MigrationTests.mm [hunk 1]
```diff
@@ -307,6 +307,25 @@ - (void)testStringPrimaryKeyNoIndexMigration {
     [RLMRealm migrateRealmAtPath:RLMTestRealmPath()];
 }
 
+- (void)testIntPrimaryKeyNoIndexMigration {
+    // make string an int
+    RLMObjectSchema *objectSchema = [RLMObjectSchema schemaForObjectClass:MigrationPrimaryKeyObject.class];
+
+    // create without search index
+    objectSchema.primaryKeyProperty.attributes = 0;
```
Comment (jpsim) [inline Realm/Tests/MigrationTests.mm]: use enum here instead of `0`?

Comment (tgoyne) [inline Realm/Tests/MigrationTests.mm]: There is no enum value for nothing.

**Comment by jpsim** [comment]
Hurray! :+1:

File: Realm/Tests/MigrationTests.mm (hunk 1 above)
Comment (jpsim) [inline Realm/Tests/MigrationTests.mm]: Ah, that's right. well that's the best we can do until we move away from `RLMPropertyAttributes` altogether (#935).

Comment (alazier) [inline Realm/Tests/MigrationTests.mm]: ```RLMPropertyAttributes``` is gone on al-swift. Once we finish up with that we are good to go. Could also just merge those changes to master if we want them out sooner but doesn't seem worth the effort yet.

File: Realm/RLMMigration.mm [hunk 2]
```diff
@@ -100,21 +100,22 @@ - (void)verifyPrimaryKeyUniqueness {
...
-                if (!table->has_search_index(primaryProperty.column)) {
-                    table->add_search_index(primaryProperty.column);
-                }
                 if (table->get_distinct_view(primaryProperty.column).size() != count) {
                     NSString *reason = [NSString stringWithFormat:@"Primary key property '%@' has duplicate values after migration.", primaryProperty.name];
                     @throw [NSException exceptionWithName:@"RLMException" reason:reason userInfo:nil];
                 }
             }
             else {
+                // FIXME: replace with count of distinct once that's supported for more than string
```
Comment (alazier) [inline Realm/RLMMigration.mm]: You tried this for int and it didn't work?

File: Realm/Tests/MigrationTests.mm [hunk 3]
```diff
@@ -307,6 +307,38 @@ - (void)testStringPrimaryKeyNoIndexMigration {
...
+
+    // create realm with old schema and populate
+    @autoreleasepool {
+        RLMRealm *realm = [self realmWithSingleObject:objectSchema];
+        [realm beginWriteTransaction];
+        [realm createObject:MigrationPrimaryKeyObject.className withObject:@[@1]];
+        [realm createObject:MigrationPrimaryKeyObject.className withObject:@[@2]];
+        [realm commitWriteTransaction];
+
+        XCTAssertFalse(realm.schema[MigrationPrimaryKeyObject.className].table->has_search_index(0));
```
Comment (alazier) [inline Realm/Tests/MigrationTests.mm]: I would prefer the pattern where we have a block and XCTAssertThrows. Every time I see this pattern it takes me a minute to figure out why we XCTAssertFalse which is annoying.

**Comment by alazier** [comment]
:+1:

File: Realm/RLMMigration.mm (hunk 2 above)
Comment (tgoyne) [inline Realm/RLMMigration.mm]: It was implemented after the most recent core release.

**Pull request merged**